ogr/ogrsf_frmts/dgn/ogrdgnlayer.cpp

```diff
@@ -528,8 +528,31 @@ OGRFeature *OGRDGNLayer::ElementToFeature( DGNElemCore *psElement, int nRecLevel
 
     strcat( szPen, ")" );
 
+    if ( CSLTestBoolean(CPLGetConfigOption("DGN_CELL_HEADER_ORIGIN", "FALSE")) && psElement->stype != DGNST_CELL_HEADER && psElement->complex ) 
```

Comment:
testing a config option can be somewhat costly. It would be better to store the value of CSLTestBoolean(...) as a member variable.
Same applies for DGNv8

Reply:
> testing a config option can be somewhat costly. It would be better to store the value of CSLTestBoolean(...) as a member variable. Same applies for DGNv8

Good point. Done!